java/core/src/test/org/apache/orc/TestOrcFilterContext.java

```diff
@@ -225,4 +247,101 @@ public void testRepeatingVector() {
+                    .fileSystem(fileSystem)
+                    .overwrite(true)
+                    .rowIndexStride(8192)
+                    .setSchema(acidSchema))) {
+
+      Random random = new Random(1024);
+      VectorizedRowBatch b = acidSchema.createRowBatch();
+      for (int rowId = 0; rowId < RowCount; rowId++) {
+        long v = random.nextLong();
+        populateColumnValues(acidSchema, b.cols,b.size, v);
+        // Populate the rowId
+        ((LongColumnVector) b.cols[3]).vector[b.size] = rowId;
+        StructColumnVector row = (StructColumnVector) b.cols[5];
+        ((LongColumnVector) row.fields[0]).vector[b.size] = rowId;
+        b.size += 1;
+        if (b.size == b.getMaxSize()) {
+          writer.addRowBatch(b);
+          b.reset();
+        }
+      }
+      if (b.size > 0) {
+        writer.addRowBatch(b);
+        b.reset();
+      }
+    }
+  }
+  
+  private void populateColumnValues(TypeDescription typeDescription, ColumnVector[] columnVectors, int size, long value) {
+    for (int columnId = 0; columnId < typeDescription.getChildren().size() ; columnId++) {
+      switch (typeDescription.getChildren().get(columnId).getCategory()) {
+        case INT:
+          ((LongColumnVector)columnVectors[columnId]).vector[size] = value;
+          break;
+        case LONG:
+          ((LongColumnVector)columnVectors[columnId]).vector[size] = value;
+          break;
+        case DECIMAL:
+          HiveDecimalWritable hiveDecimalWritable = new HiveDecimalWritable();
+          hiveDecimalWritable.setFromLongAndScale(value, scale);
+          ((DecimalColumnVector) columnVectors[columnId]).vector[size] = hiveDecimalWritable;
+          break;
+        case STRING:
+          ((BytesColumnVector) columnVectors[columnId]).setVal(size,
+                  String.valueOf(value)
+                          .getBytes(StandardCharsets.UTF_8));
+          break;
+        case STRUCT:
+          populateColumnValues(typeDescription.getChildren().get(columnId), ((StructColumnVector)columnVectors[columnId]).fields, size, value);
+          break;           
+        default:
+          throw new IllegalArgumentException();
+      }
+    }
+  }
+  private void readSingleRowWithFilter(long id) throws IOException {
+    Reader reader = OrcFile.createReader(filePath, OrcFile.readerOptions(configuration).filesystem(fileSystem));
+    SearchArgument searchArgument = SearchArgumentFactory.newBuilder()
+            .in("int1", PredicateLeaf.Type.LONG, id)
+            .build();
+    Reader.Options readerOptions = reader.options()
+            .searchArgument(searchArgument, new String[] {"int1"})
+            .useSelected(true)
+            .allowSARGToFilter(true);
+    VectorizedRowBatch vectorizedRowBatch = acidSchema.createRowBatch();
+    long rowCount = 0;
+    try (RecordReader recordReader = reader.rows(readerOptions)) {
+      assertTrue(recordReader.nextBatch(vectorizedRowBatch));
+      rowCount += vectorizedRowBatch.size;
```

Comment:
Can we please validate that the entry is matching? i.e. the value of the LongColumnVector value is the `id` that was passed in.

Reply:
Added more validation.